BDArmory/Ammo/PooledRocket.cs

```diff
@@ -718,7 +718,7 @@ void Detonate(Vector3 pos, bool missed, Part PenetratingHit = null)
                         Vector3 direction = default(Vector3);
                         if (shaped)
                         {
-                            direction = (pos + rb.velocity * Time.deltaTime).normalized;
+                            direction = (pos + (rb.velocity * Time.deltaTime)).normalized;
```

Comment:
This is still a position. If you want the velocity direction, then it should simply be
```
direction = rb.velocity.normalized;
```
otherwise it should be based on `transform.forward` (or `up` depending on the model).

Reply:
transform.forward would be ideal, but we have no guarantee that mod rockets would have proper transform orientations (heck, the stock rockets aren't consistent in that regard)

Reply:
Thinking a little more about this, we may need to add in the Krakensbane frame velocity, i.e.,
```csharp
direction = (rb.velocity + Krakensbane.GetFrameVelocityV3f()).normalized;
```

Also, I should benchmark the `Krakensbane.GetFrameVelocityV3f()` function at some point and see whether it ought to be cached. It shouldn't cause GC allocation since `Vector3` is a `struct` and thus goes on the stack, but there may be some performance overhead in the function that can be avoided.

Reply:
Actually, scratch that last comment, `direction = rb.velocity.normalized` should be correct here.

Looking through the PooledRocket.cs code a bit more, I see that the `rb.velocity` is handled differently from in PooledBullet.cs. There, `currentVelocity` is always the real velocity without KB offloading (except sub-projectiles look to be getting an extra frame velocity added on line 1267), but this doesn't look to be the case in PooledRocket.cs. By the looks of it, the `rb.velocity` is being set to the real velocity of the parent rigid body to start with and KB offloading isn't being applied to the rockets (I guess this is due to KSP not applying KB corrections to them since they're not vessels). There's a couple of places where the KB frame velocity is being added though, which it probably shouldn't be (lines 283 and 901).

Reply:
I've corrected the frame velocity for beehive sub-projectiles and the rocket model orientation in commit 513f7752.
It also optimises access to the `Krakensbane` and `FloatingOrigin` adjustments (~4x speed up, though they're already small so it probably won't have a large effect overall: 1.44E-08s → 2.97E-09s).